web/docs/auth/social-auth/keycloak.md

```diff
@@ -0,0 +1,547 @@
+
+    - Once you know on which URL(s) your API server will be deployed, also add those URL(s).
+    - For example: `https://my-server-url.com/auth/keycloak/callback`.
+
+1. Click "Save".
+1. In the "Credentials" tab, copy the "Client Secret" value, which we'll use in the next step.
+
+  ![Keycloak Screenshot 5](/img/auth/keycloak/5-keycloak.png)
+
+### 4. Adding Environment Variables
+
+Add these environment variables to the `.env.server` file at the root of your project (take their values from the previous step):
+
+```bash title=".env.server"
+KEYCLOAK_CLIENT_ID=your-keycloak-client-id
+KEYCLOAK_CLIENT_SECRET=your-keycloak-client-secret
+KEYCLOAK_REALM_URL=https://your-keycloak-url.com/realms/master
+```
+
+We assumed in the `KEYCLOAK_REALM_URL` env variable that you are using the `master` realm. If you are using a different realm, replace `master` with your realm name.
+
+### 5. Adding the Necessary Routes and Pages
+
+Let's define the necessary authentication Routes and Pages.
+
+Add the following code to your `main.wasp` file:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+// ...
+
+// 6. Define the routes
+route LoginRoute { path: "/login", to: LoginPage }
+page LoginPage {
+  component: import { Login } from "@src/pages/auth.jsx"
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+// ...
+
+// 6. Define the routes
+route LoginRoute { path: "/login", to: LoginPage }
+page LoginPage {
+  component: import { Login } from "@src/pages/auth.tsx"
+}
+```
+
+</TabItem>
+</Tabs>
+
+We'll define the React components for these pages in the `src/pages/auth.{jsx,tsx}` file below.
+
+### 6. Create the Client Pages
+
+:::info
+We are using [Tailwind CSS](https://tailwindcss.com/) to style the pages. Read more about how to add it [here](../../project/css-frameworks).
+:::
+
+Let's now create a `auth.{jsx,tsx}` file in the `src/pages`.
+It should have the following code:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```tsx title="src/pages/auth.jsx"
+import { LoginForm } from 'wasp/client/auth'
+
+export function Login() {
+  return (
+    <Layout>
+      <LoginForm />
+    </Layout>
+  )
+}
+
+// A layout component to center the content
+export function Layout({ children }) {
+  return (
+    <div className="w-full h-full bg-white">
+      <div className="min-w-full min-h-[75vh] flex items-center justify-center">
+        <div className="w-full h-full max-w-sm p-5 bg-white">
+          <div>{children}</div>
+        </div>
+      </div>
+    </div>
+  )
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```tsx title="src/pages/auth.tsx"
+import { LoginForm } from 'wasp/client/auth'
+
+export function Login() {
+  return (
+    <Layout>
+      <LoginForm />
+    </Layout>
+  )
+}
+
+// A layout component to center the content
+export function Layout({ children }: { children: React.ReactNode }) {
```

Comment:
I think you should import `React`.

Reply:
Types work just fine even without importing it, I've just tested it

Reply:
Hm, strange. But if everything works (hope you checked build), then no problem.

I'd still probably do:
```typescript
import { ReactNode } from 'react';
```
But it's just a matter of personal style.